Update docs on using concurrency to auto-cancel redundant jobs / runs (#11244)
github/docs: data/reusables/actions/actions-group-concurrency.md

  cancel-in-progress: true
```
{% endraw %}

### Example: Using a fallback value

If you build the group name with a property that is only defined for specific events, you can use a fallback value. For example, `github.head_ref` is only defined on `pull_request` events. If your workflow responds to other events in addition to `pull_request` events, you will need to provide a fallback to avoid a syntax error. The following concurrency group cancels in-progress jobs or runs on `pull_request` events only; if `github.head_ref` is undefined, the concurrency group will fallback to the run ID, which is guaranteed to be both unique and defined for the run.

{% raw %}
```yaml
concurrency: 
  group: ${{ github.head_ref || github.run_id }}
  cancel-in-progress: true
```
{% endraw %}

> ### Only cancel in-progress jobs or runs for the current workflow

Reviewer: ```suggestion
### Example: Only cancel in-progress jobs or runs for the current workflow
```

  cancel-in-progress: true
```
{% endraw %}

{% note %}

**Note:** `github.head_ref` is only defined on `pull_request` events. If you have a workflow that responds to other events in addition to `pull_request` events, you will need to provide a fallback to avoid a syntax error.

{% endnote %}

### Example: With fallback

To cancel in-progress jobs or runs on `pull_request` events only, but not on other events the workflow responds to:

{% raw %}
```yaml
concurrency: 
  group: ${{ github.head_ref || github.run_id }}
  cancel-in-progress: true
```
{% endraw %}

If `github.head_ref` is undefined, the concurrency group will fallback to the run ID, which is guaranteed to be both unique and defined for the run.

### Only cancel in-progress jobs or runs for the current workflow

{% note %}

**Note:** If you have multiple workflows in the same repository, concurrency group names must be unique to avoid canceling in-progress jobs or runs from independent workflows. Otherwise, any previously in-progress or pending job will be canceled, regardless of the workflow that triggered it, leading to potentially confusing behavior if that wasn't your intent.

{% endnote %}

> To only cancel in-progress runs on a per-workflow basis:

Reviewer: ```suggestion
To only cancel in-progress runs of the same workflow, you can use the `github.workflow` property to build the concurrency group:
```

{% raw %}
```yaml
concurrency: 

>   group: ${{ github.workflow }}-${{ github.head_ref }}

Reviewer: to make it more generic:
```suggestion
  group: ${{ github.workflow }}-${{ github.ref }}
```

  cancel-in-progress: true
```
{% endraw %}

> With fallback:
> 
> {% raw %}
> ```yaml
> concurrency: 
>   group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
>   cancel-in-progress: true
> ```
> {% endraw %}

Reviewer: Since we demo the fallback above, I think we can omit this example here
```suggestion
```
